pkg/cloud/amazon/aws_kms.go

```diff
@@ -159,9 +233,39 @@ func (k *awsKMS) MasterKeyID() (string, error) {
 
 // Encrypt implements the KMS interface.
 func (k *awsKMS) Encrypt(ctx context.Context, data []byte) ([]byte, error) {
+	var grantTokens []*string
```

Comment:
For these grant operations, do we think we can create the grant at instantiation of `awsKMS` and then retire it at Close()? It seems a little bit suspicious that we have to create and delete the grant after every encrypt

Reply:
Hmm so the reason why i put it in the encrypt/decrypt function itself is that I was thinking a user might only want to backup only so might only pass in a role with encrypt permissions only.

Creating the grant when awsKMS is created would require both permissions.

A workaround might be to just attempt to create one grant for encrypt and one for decrypt? And then as long as one worked don't error out in the instantiation but rather when we call encrypt/decrypt?

Or maybe it's not a problem at all that we force the user to give us both encrypt and decrypt always?

Reply:
I think I actually just misunderstood create Grant in general. The docs say something about how it goes alongside IAM policies but I think that actually means that you can use it instead of IAM policies. Rewriting it so that it just uses IAM Assume Role and so far it seems like it works